### ISSUE TITLE
Local version shows red "saving" bar

### ISSUE THREAD

**Issue by smellymoo** [mpcjanssen/simpletask-android#1192]
When you use the local file version, it still shows the thin red line at the top to denote saving, it lasts 2-3 seconds, makes no sense.

**Comment by sphenotux** [comment]
Hm... I see the red line from time to time. I guess (please correct me, long-term users)
* You change your todo.txt locally (on your smartphone?)
* Changes are saved to the local todo.txt on your device
* App notices those changes and tries to sync them to the server
* if that sync fails, red line is shown
@smellymoo you could crosscheck (please)
* e. g. via file browser on your device if the timestamp of your local file corresponds to your latest (local) changes
* what happens if you have a stable, reliable server connection
* what happens if you truly disconnect, e.g. in flight mode, and make changes
Details would make me happy

**Comment by smellymoo** [comment]
I mean the server-less version of simpletask, it isn't supposed to be using the internet, it just saves a local file. But it still has a red saving bar.

**Comment by sphenotux** [comment]
Weird...
Can you please cross check read/write permissions of the folder the todo.txt file is located in? Some users experienced success here.

**Comment by smellymoo** [comment]
it saves it, so it's not getting blocked by permissions, although I haven't checked yet. It's like it has a timer in the save function for displaying the red bar.

**Comment by sphenotux** [comment]
I use Nextcloud version 11 and rarely see the red bar. Are you running the latest version? 
Check: Main menu -  settings - miscellaneous (or similar) -  error log and compare to the latest version in your trusted store. 
How long is the bar shown:
* 1/10 second
* 3 seconds
* 2 minutes?

Hard to tell for me and maybe an inpolite question, but if changes are reliably saved: Can you live with it?

**Comment by sphenotux** [comment]
... in addition: Main menu - settings - functions (or similar): '*Time before save*' has **a value - which**? Mine is set to three seconds.

**Comment by smellymoo** [comment]
Yup! It's that. Which means it only needs a slight correction for the local version (set timeout to 0, which you can't do).

**Comment by sphenotux** [comment]
Thx, so you think the **default value** is > 0 (saying: greater than zero) now?

**Comment by mpcjanssen** [comment]
The timer for the local version doesn't really make sense, it is useful for the connected versions to wait for the todo.txt to settle before saving it. I think setting different defaults for local and connected versions should be relatively easy.

**Comment by smichel17** [comment]
In the past I have synced the local version using the nextcloud app (and actually found it handled conflict resolution better than the ST Nextcloud flavor).

Not sure if waiting would help anything in that situation, but just as an FYI.

**Comment by mpcjanssen** [comment]
The waiting is mostly to not trigger throttling of APIs and to prevent conficts on the remote side by starting a new save without waiting for the result of the previous one. Saving to local storage is almost instantanious so it's not a real issue there.
